it/v3/endpoints/RetrieveCodingOutStatusControllerISpec.scala

```diff
@@ -0,0 +1,144 @@
+
+      s"validation fails with ${expectedBody.code} error" in new Test {
+        override protected val nino: String    = requestNino
+        override protected val taxYear: String = requestTaxYear
+
+        override def setupStubs(): StubMapping = {
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+        }
+
+        val response: WSResponse = await(request.get())
+        response.status shouldBe expectedStatus
+        response.json shouldBe Json.toJson(expectedBody)
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    val input = List(
+      ("AA1123A", "2021-22", BAD_REQUEST, NinoFormatError),
+      ("AA123456A", "20199", BAD_REQUEST, TaxYearFormatError)
+    )
+    input foreach (validationErrorTest _).tupled
+  }
+
+  "downstream service error" should {
+    def serviceErrorTest(downstreamStatus: Int, downstreamCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+      s"handle $downstreamCode with status $downstreamStatus" in new Test {
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DownstreamStub.onError(DownstreamStub.GET, downstreamUrl, downstreamStatus, errorBody(downstreamCode))
+        }
+
+        val response: WSResponse = await(request.get())
+        response.status shouldBe expectedStatus
+        response.json shouldBe Json.toJson(expectedBody)
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    val input = List(
+      (BAD_REQUEST, "INVALID_TAXABLE_ENTITY_ID", BAD_REQUEST, NinoFormatError),
+      (BAD_REQUEST, "INVALID_TAX_YEAR", BAD_REQUEST, TaxYearFormatError),
+      (BAD_REQUEST, "INVALID_REGIME", INTERNAL_SERVER_ERROR, InternalError),
+      (BAD_REQUEST, "INVALID_CORRELATIONID", INTERNAL_SERVER_ERROR, InternalError),
+      (BAD_REQUEST, "DUPLICATE_SUBMISSION", INTERNAL_SERVER_ERROR, InternalError),
+      (BAD_REQUEST, "BUSINESS_PARTNER_NOT_EXIST", BAD_REQUEST, BusinessPartnerNotExistError),
+      (BAD_REQUEST, "ITSA_CONTRACT_OBJECT_NOT_EXIST", BAD_REQUEST, ITSAContractObjectNotExistError),
+      (BAD_REQUEST, "REQUEST_NOT_PROCESSED", INTERNAL_SERVER_ERROR, InternalError),
+      (INTERNAL_SERVER_ERROR, "SERVER_ERROR", INTERNAL_SERVER_ERROR, InternalError),
+      (BAD_REQUEST, "BAD_GATEWAY", INTERNAL_SERVER_ERROR, InternalError),
+      (BAD_REQUEST, "SERVICE_UNAVAILABLE", INTERNAL_SERVER_ERROR, InternalError)
+    )
+    input.foreach(args => (serviceErrorTest _).tupled(args))
+  }
+
+  private trait Test {
+    protected val nino    = "AA123456A"
+    protected val taxYear = "2023-24"
+
+    def downstreamUrl: String = {
+      s"/income-tax/accounts/self-assessment/tax-code/opt-out/itsa/$nino/${TaxYear.fromMtd(taxYear).year}"
```

Comment:
Maybe be more explicit about the expected tax year in the url (since we're expecting it to be re-formatted):
```suggestion
      s"/income-tax/accounts/self-assessment/tax-code/opt-out/itsa/$nino/2024"
```